src/manifest_parser.cc

```diff
@@ -408,14 +406,14 @@ bool ManifestParser::ParseFileInclude(bool new_scope, string* err) {
     return false;
   string path = eval.Evaluate(env_);
 
-  ManifestParser subparser(state_, file_reader_, options_);
+  ManifestParser *subparser = new ManifestParser(state_, file_reader_, options_);
```

Comment:
needs to be fixed.

Reply:
but if we follow the ninja approach we leave it to the OS to free?

Reply:
> Overall, I think this changes way too many things all at the same time.
> 
> I recommend breaking these changes into smaller pieces, and submitting them individually.
You could maybe  review/accept/reject the first commit in the pull request, which is smaller?